src/window_selectable.cpp

```diff
@@ -127,13 +127,15 @@ void Window_Selectable::Update() {
 	Window_Base::Update();
 	if (active && item_max > 0 && index >= 0) {
 		if (Input::IsRepeated(Input::DOWN) || Input::IsTriggered(Input::SCROLL_DOWN)) {
-			if (index < item_max - column_max || column_max == 1) {
+			if (index < item_max - column_max || (!disable_endless_scrolling && column_max == 1) || (column_max == 1 &&
+				(Input::IsTriggered(Input::DOWN) || Input::IsTriggered(Input::SCROLL_DOWN)))) {
 				Main_Data::game_system->SePlay(Main_Data::game_system->GetSystemSE(Main_Data::game_system->SFX_Cursor));
 				index = (index + column_max) % item_max;
 			}
 		}
 		if (Input::IsRepeated(Input::UP) || Input::IsTriggered(Input::SCROLL_UP)) {
-			if (index >= column_max || column_max == 1) {
+			if (index >= column_max || (!disable_endless_scrolling && column_max == 1) || (column_max == 1 &&
+				(Input::IsTriggered(Input::UP) || Input::IsTriggered(Input::SCROLL_UP)))) {
```

Comment:
You check twice for IsTriggered which was already done 2 lines earlier. (same for DOWN)
Why is that necessary?

Reply:
Because if endless scrolling is disabled, the second check ensures that you still can jump to the begin or end of the menu if you press up or down while not holding the key down.

Reply:
Imo this if-stuff is quite ugly. Alternative patch (apply this to your HEAD and then squash this commit and my patch together): https://gist.github.com/Ghabry/cc1a20c3b31f461a5cf2a51e1519b9ab

Changes:
``disable_endless_scrolling`` is now ``endless_scrolling`` because double negation is bad (``!disable_endless_scrolling -> endless_scrolling``).
The move up and down logic is extracted in a lambda and the branches switched between "triggered" and "repeating".

Please test if I missed any case for "endless_scrolling = false".

Reply:
Thanks for the patch, it works for me. It is included now.